src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,200 @@
+import duke.logic.Parser;
+import duke.logic.Storage;
+import duke.logic.TaskList;
+import duke.logic.Ui;
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+/**
+ * Encapsulates the main Duke programme.
+ */
+public class Duke extends Application {
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    private Image user = new Image(Objects.requireNonNull(this.getClass().getResourceAsStream(
+            "/images/default.png")));
+    private Image duke = new Image(Objects.requireNonNull(this.getClass().getResourceAsStream(
+            "/images/monke.png")));
+
+    /**
+     * Storage responsible for reading from and writing to local file.
+     */
+    private Storage storage;
+
+    /**
+     * List of tasks stored at run-time.
+     */
+    private TaskList taskList;
+
+    /**
+     * UI responsible for displaying messages to user.
+     */
+    private Ui ui;
+
+    /**
+     * Default constructor for an instance of Duke. Used for GUI.
+     */
+    public Duke() {
+        try {
+            this.ui = new Ui();
+            this.storage = new Storage("./data/saved.txt");
+            this.taskList = new TaskList(storage.readFromFile());
+        } catch (DukeException e) {
+            ui.showError(e);
+            this.taskList = new TaskList();
+        }
+    }
+
+    /**
+     * Constructor for an instance of Duke.
+     *
+     * @param filePath Path at which local file is stored.
+     */
+    public Duke(String filePath) {
+        try {
+            this.ui = new Ui();
+            this.storage = new Storage(filePath);
+            this.taskList = new TaskList(storage.readFromFile());
+        } catch (DukeException e) {
+            ui.showError(e);
+            this.taskList = new TaskList();
+        }
+    }
+
+    /**
+     * Boots up instance of Duke programme.
+     */
+    public void run() {
+        this.ui.openScanner();
+        this.ui.showIntro();
+        boolean hasExited = false;
+
+        do {
+            try {
+                String fullCommand = ui.readCommand();
+                Command command = Parser.parse(fullCommand);
+                command.execute(this.taskList, this.ui, this.storage);
+                hasExited = command.isExitCommand();
+            } catch (DukeException e) {
+                ui.showError(e);
+            }
+        } while (!hasExited);
+
+        ui.closeScanner();
+    }
+
+    @Override
+    public void start(Stage stage) {
```

Comment:
Could this method be shortened by abstracting the initialisation of different items into methods like `setStage`, `setScrollPane`, `setAnchorPane` and calling them here instead?